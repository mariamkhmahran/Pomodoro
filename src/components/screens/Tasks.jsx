import React from "react";
import localforage from "localforage";
import tasks from "src/mocks/TasksData.json";
import Header from "src/components/layout/Header";
import { TaskCard } from "src/components/atoms/Card";

export default class Tasks extends React.Component {
  state = {
    tasks: [],
    done: []
  };

  componentDidMount() {
    let tempTasks = [],
      tempDone = [];
    tasks.data.map(task => {
      task.done ? tempDone.push(task) : tempTasks.push(task);
    });
    this.setState(
      () => ({
        tasks: tempTasks,
        done: tempDone
      }),
      () => {
        localforage.setItem("tasks", this.state.tasks, () => {
          localforage.setItem("done", this.state.done);
        });
      }
    );
  }

  // componentDidMount() {
  //   this.updateState();
  // }

  updateState = () => {
    localforage.getItem("tasks", (err, tasks) => {
      this.setState(
        () => ({ tasks: tasks }),
        () => {
          localforage.getItem("done", (err, done) => {
            this.setState(() => ({ done: done }));
          });
        }
      );
    });
  };

  renderCard(task) {
    return <TaskCard task={task} onclick={this.markAsDone} />;
  }

  addTask = () => {
    let task = {
      id:
        "_" +
        Math.random()
          .toString(36)
          .substr(2, 9),
      name: document.getElementById("name").value,
      cyclesDone: 0,
      estimatedTime: document.getElementById("cycles").value,
      deadline: document.getElementById("deadline").value,
      subTasks: [],
      done: false
    };

    let self = this;

    localforage.getItem("tasks", (err, tasks) => {
      debugger;
      localforage.setItem("tasks", [task, ...tasks], () => {
        self.updateState();
        self.clearFields();
        var modal = document.getElementById("modal");
        modal.style.display = "none";
      });
    });
  };

  clearFields() {
    document.getElementById("name").value = "";
    document.getElementById("cycles").value = "";
    document.getElementById("deadline").value = null;
  }

  markAsDone = task => {
    let index = this.state.tasks.indexOf(task);
    if (index > -1) {
      let tempTask = this.state.tasks;
      tempTask.splice(index, 1);
      task.done = true;
      task.subTasks.forEach(task => {
        task.done = true;
      });
      this.setState(
        ({ tasks, done }) => ({
          tasks: tempTask,
          done: [task, ...done]
        }),
        () => {
          localforage.setItem("tasks", this.state.tasks, () => {
            localforage.setItem("done", this.state.done);
          });
        }
      );
    }
  };

  Modal = () => (
    <div id="modal" className="modal">
      <div className="modal-content">
        <span className="close">&times;</span>
        <span className="subtitle orange">ADD NEW TASK</span>
        <input
          className="inputField"
          type="text"
          id="name"
          placeholder="Name"
        />
        <input
          className="inputField"
          type="text"
          id="cycles"
          placeholder="Estimated total number of cycles"
        />
        <input
          className="inputField"
          type="date"
          id="deadline"
          placeholder="Deadline"
        />
        <div>
          <button className="btn addTask" onClick={this.addTask}>
            Add Task
          </button>
        </div>
      </div>
    </div>
  );

  showModal = () => {
    var modal = document.getElementById("modal");
    modal.style.display = "block";

    var span = document.getElementsByClassName("close")[0];
    let self = this;
    span.onclick = function() {
      modal.style.display = "none";
      self.clearFields();
    };

    window.onclick = function(event) {
      if (event.target === modal) {
        modal.style.display = "none";
      }
    };
  };

  render() {
    return (
      <div className="router_output">
        <Header />
        <this.Modal />
        <div className="subHeader">
          <span className="subtitle">TASKS</span>
          <button className="btn plusBtn" onClick={this.showModal}>
            <i className="fa fa-plus" />
          </button>
        </div>
        <div className="tasks">
          <div>
            <div className="col">
              <div className="transparent_box">
                <span className="subtitle">TO DO</span>
                <div className="cardList">
                  {this.state.tasks.map(task => this.renderCard(task))}
                </div>
              </div>
            </div>
            <div className="col">
              <div className="transparent_box">
                <span className="subtitle">DONE</span>
                <div className="cardList">
                  {this.state.done.map(task => this.renderCard(task))}
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
