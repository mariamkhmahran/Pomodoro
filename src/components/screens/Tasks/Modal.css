/* Modal styles*/
.modal {
  display: none;
  position: fixed;
  z-index: 3;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  align-items: center;
  justify-content: center;
}
.modal[data-is-open="true"] {
  display: flex;
}

.modal__screen {
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background-color: rgb(0, 0, 0);
  background-color: rgba(0, 0, 0, 0.4);
}
.modal__content {
  background-color: #fefefe;
  padding: 20px;
  border: 1px solid #888;
  border-radius: 10px;
  width: 35%;
  animation-name: modalOpen;
  animation-duration: 0.5s;
  z-index: 4;
}

.addTask {
  background: #f2a969;
  border-radius: 50px;
  color: white;
  font-size: 30px;
  padding: 5px 20px;
  margin: 10px;
}

@keyframes modalOpen {
  from {
    display: none;
    opacity: 0;
  }
  to {
    display: flex;
    opacity: 1;
  }
}

.modal__close {
  color: #aaa;
  float: right;
  font-size: 28px;
  font-weight: bold;
}

.modal__close:hover,
.modal__close:focus {
  color: black;
  text-decoration: none;
  cursor: pointer;
}

/* SNACKBAR */

#snackbar {
  visibility: hidden; /* Hidden by default. Visible on click */
  min-width: 250px; /* Set a default minimum width */
  margin-left: -125px; /* Divide value of min-width by 2 */
  background-color: white; /* Black background color */
  color: #857261; /* White text color */
  text-align: center; /* Centered text */
  border-radius: 10px; /* Rounded borders */
  padding: 16px; /* Padding */
  position: fixed; /* Sit on top of the screen */
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  z-index: 5; /* Add a z-index if needed */
  left: 50%; /* Center the snackbar */
  top: 50px; /* 30px from the bottom */
}

/* Show the snackbar when clicking on a button (class added with JavaScript) */
#snackbar.show {
  visibility: visible; /* Show the snackbar */

  /* Add animation: Take 0.5 seconds to fade in and out the snackbar. 
However, delay the fade out process for 2.5 seconds */
  -webkit-animation: fadein 0.5s, fadeout 0.5s 2.5s;
  animation: fadein 0.5s, fadeout 0.5s 2.5s;
}

/* Animations to fade the snackbar in and out */
@-webkit-keyframes fadein {
  from {
    top: 0;
    opacity: 0;
  }
  to {
    top: 50px;
    opacity: 1;
  }
}

@keyframes fadein {
  from {
    top: 0;
    opacity: 0;
  }
  to {
    top: 50px;
    opacity: 1;
  }
}

@-webkit-keyframes fadeout {
  from {
    top: 50px;
    opacity: 1;
  }
  to {
    top: 0;
    opacity: 0;
  }
}

@keyframes fadeout {
  from {
    top: 50px;
    opacity: 1;
  }
  to {
    top: 0;
    opacity: 0;
  }
}

@media screen and (max-width: 800px) {
  .modal__content {
    padding: 20px;
    width: 99%;
    height: 80%;
  }
}
