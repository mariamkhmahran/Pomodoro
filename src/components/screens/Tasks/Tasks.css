.col {
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 90%;
  width: 50%;
}

.tasks > div:first-child {
  display: flex;
  flex-wrap: wrap;
  width: 100%;
  padding: 0px 96px;
}

.tasks .transparent_box {
  margin: 2%;
  padding: 2%;
  width: 93%;
  height: 600px;
}

.cardList {
  height: fit-content;
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  overflow-y: scroll;
  overflow-x: hidden;
}

.plusBtn {
  color: #f2a969;
  font-size: 32px;
  height: 50px;
  width: 70px;
  border-radius: 39px;
}
.plusBtn:hover {
  background-color: rgb(249, 249, 249);
  box-shadow: 1px 0px 15px #ce8f58;
}

@media screen and (max-width: 800px) {
  .col {
    height: 90vh;
    width: 100%;
  }
  .tasks .transparent_box {
    height: 80vh;
  }
  .tasks > div:first-child {
    padding: 0px 6px;
  }
}
