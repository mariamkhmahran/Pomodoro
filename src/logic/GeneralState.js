import localforage from "localforage";

async function set(attr, value, callback) {
  localforage.setItem(attr, value, callback);
  return await get(attr);
}

async function get(attr, callback) {
  return await localforage.getItem(attr, callback);
}

async function getIndex(attr, item, func) {
  let array = await get(attr);
  let index = array.findIndex(func);
  return index;
}

async function getAll(callback) {
  let tasks = await get("tasks");
  let done = await get("done");
  let queue = await get("queue");
  let cycles = await get("cycles");

  if (callback) {
    Promise.all([tasks, done, queue, cycles]).then(
      callback({ tasks, queue, done, cycles })
    );
  }
  return { tasks, done, queue, cycles };
}

async function removeItem(attr, func) {
  let array = await get(attr);
  let index = array.findIndex(func);
  if (~index) {
    array.splice(index, 1);
    await set(attr, array);
    return array;
  }
}

async function addAtBeginning(attr, item) {
  let array = await get(attr);
  await localforage.setItem(attr, [item, ...array]);
}

async function addAtEnd(attr, item) {
  let array = await get(attr);
  await localforage.setItem(attr, [...array, item]);
}

async function updateItem(attr, item, func) {
  let array = await get(attr);
  let index = array.findIndex(func);
  array.splice(index, 1, item);
  set(attr, array);
}

export { set, get, getIndex, getAll, removeItem, addAtBeginning, addAtEnd, updateItem };
