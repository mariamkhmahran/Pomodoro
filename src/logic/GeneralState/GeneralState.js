import localforage from "localforage";

async function set(attr, value, callback) {
  await localforage.setItem(attr, value, callback);
  return value;
}

async function get(attr, callback) {
  return await localforage.getItem(attr, callback);
}

async function getIndex(attr, func) {
  let array = await get(attr);
  let index = array.findIndex(func);
  return index;
}

async function getAll(callback) {
  let tasks = await get("tasks");
  let done = await get("done");
  let queue = await get("queue");
  let cycles = await get("cycles");

  if (callback) {
    Promise.all([tasks, done, queue, cycles]).then(
      callback({ tasks, queue, done, cycles })
    );
  }
  return { tasks, done, queue, cycles };
}

async function removeItem(attr, func) {
  let array = await get(attr);
  let index = await getIndex(attr, func);
  if (~index) {
    array.splice(index, 1);
    await set(attr, array);
    return array;
  }
}

async function addAtBeginning(attr, item) {
  let array = await get(attr);
  await localforage.setItem(attr, [item, ...array]);
}

async function addAtEnd(attr, item) {
  let array = await get(attr);
  await localforage.setItem(attr, [...array, item]);
}

async function updateItem(attr, item, func) {
  let index = await getIndex(attr, func);
  let array;
  if (~index) {
    array = await get(attr);
    array.splice(index, 1, item);
    await set(attr, array);
  }
}

async function reOrder(queue, startIndex, endIndex) {
  if (startIndex > -1 && endIndex < queue.length) {
    let result = [...queue];
    const [removed] = result.splice(startIndex, 1);
    result.splice(endIndex, 0, removed);
    set("queue", result);
    return result;
  }
  return null;
}

export {
  set,
  get,
  getIndex,
  getAll,
  removeItem,
  addAtBeginning,
  addAtEnd,
  updateItem,
  reOrder
};
