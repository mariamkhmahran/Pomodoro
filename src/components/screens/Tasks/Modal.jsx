import React from "react";

import "./Modal.css";

export default class Modal extends React.Component {
  onClose = () => {
    this.clearFields();
    this.props.onClose();
  };

  clearFields() {
    document.getElementById("name").value = "";
    document.getElementById("cycles").value = "";
    document.getElementById("deadline").value = null;
  }

  addTask = () => {
    if (
      document.querySelector("#name").value === "" ||
      document.querySelector("#cycles").value === "" ||
      document.querySelector("#deadline").value === null
    ) {
      this.showSnackbar();
      return;
    }

    let name = document.querySelector("#name").value,
      cycles = document.querySelector("#cycles").value,
      deadline = document.querySelector("#deadline").value;

    this.props.addTask({ name, cycles, deadline });
    this.onClose();
  };

  showSnackbar() {
    var x = document.getElementById("snackbar");
    x.className = "show";
    setTimeout(function() {
      x.className = x.className.replace("show", "");
    }, 3000);
  }

  render() {
    const { isOpen } = this.props;
    const { addTask, onClose } = this;
    return (
      <div className="modal" data-is-open={isOpen}>
        <div className="modal__screen" onClick={onClose} />
        <div id="snackbar">Please fill all information..</div>
        <div className="modal__content">
          <span onClick={onClose} className="modal__close">
            &times;
          </span>
          <span className="subtitle orange">ADD NEW TASK</span>
          <input
            className="inputField"
            type="text"
            id="name"
            placeholder="Name"
          />
          <input
            className="inputField"
            type="text"
            id="cycles"
            placeholder="Estimated total number of cycles"
          />
          <input
            className="inputField"
            type="date"
            id="deadline"
            placeholder="Deadline"
          />
          <div>
            <button className="btn addTask" onClick={addTask}>
              Add Task
            </button>
          </div>
        </div>
      </div>
    );
  }
}
