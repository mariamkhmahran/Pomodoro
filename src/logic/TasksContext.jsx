import React from "react";
import * as GeneralState from "src/logic/GeneralState";

export const ContextTasks = React.createContext();

class TimerContext extends React.Component {
  state = {
    tasks: [],
    queue: [],
    done: [],
    addTask: this.addTask,
    addCycle: this.addCycle,
    markAsDone: this.markAsDone
  };
  componentDidMount() {
    this.update();
  }
  update = async () => {
    let state = await GeneralState.getAll();
    this.setState(
      () => ({
        ...state,
        addTask: this.addTask,
        addCycle: this.addCycle,
        markAsDone: this.markAsDone
      }),
      () => {
        console.log(this.state);
      }
    );
  };

  addTask = async ({ name, cycles, deadline }) => {
    let id = "_" + Math.random().toString(36);
    let task = {
      id: id.substr(2, 9),
      name: name,
      cyclesDone: 0,
      estimatedTime: cycles,
      deadline: deadline,
      subTasks: [],
      done: false
    };
    let queue = GeneralState.get("queue");

    let promise = GeneralState.set("tasks", [task, ...this.state.tasks]);
    GeneralState.set("queue", [...queue, task]);

    Promise.all([promise]).then(() => {
      this.update();
      var modal = document.getElementById("modal");
      modal.style.display = "none";
    });
  };

  addCycle = async task => {
    task.cyclesDone += 1;

    let Promises = [
      GeneralState.updateItem("task", task, t => t.id === task.id),
      GeneralState.updateItem("queue", task, t => t.id === task.id)
    ];

    Promise.all(Promises).then(() => {
      this.update();
    });
  };

  markAsDone = async task => {
    task.done = true;
    task.subTasks.forEach(task => {
      task.done = true;
    });

    let Promises = [
      GeneralState.removeItem("tasks", t => t.id === task.id),
      GeneralState.removeItem("queue", t => t.id === task.id),
      GeneralState.addAtBeginning("done", task)
    ];

    Promise.all(Promises).then(() => {
      this.update();
    });
  };

  render() {
    return (
      <ContextTasks.Provider value={this.state}>
        {this.props.children}
      </ContextTasks.Provider>
    );
  }
}
export default TimerContext;
