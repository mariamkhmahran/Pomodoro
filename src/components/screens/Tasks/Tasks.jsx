import React from "react";
import Modal from "./Modal";
import { TaskCard } from "../../atoms/Card";
import { ContextTasks } from "../../../logic/TasksContext";
import "./Tasks.css";

export default class Tasks extends React.Component {
  state = {
    addTaskModalIsOpen: false
  };

  render() {
    const { addTaskModalIsOpen } = this.state;
    return (
      <ContextTasks.Consumer>
        {({ tasks, done, addTask, addCycle, markAsDone, deleteTask }) => {
          return (
            <div className="router_output">
              <Modal
                addTask={addTask}
                isOpen={addTaskModalIsOpen}
                onClose={() => this.setState({ addTaskModalIsOpen: false })}
              />
              <div className="subHeader">
                <span className="subtitle">TASKS</span>
                <button
                  className="btn plusBtn"
                  onClick={() => this.setState({ addTaskModalIsOpen: true })}>
                  <i className="fa fa-plus" />
                </button>
              </div>
              <div className="Main tasks">
                <div>
                  <div className="col">
                    <div className="transparent_box">
                      <span className="subtitle">TO DO</span>
                      <div className="cardList">
                        {tasks.map(task => (
                          <TaskCard
                            key={task.id}
                            task={task}
                            onclick={markAsDone}
                            onDelete={deleteTask}
                          />
                        ))}
                      </div>
                    </div>
                  </div>
                  <div className="col">
                    <div className="transparent_box">
                      <span className="subtitle">DONE</span>
                      <div className="cardList">
                        {done.map(task => (
                          <TaskCard
                            key={task.id}
                            task={task}
                            onclick={markAsDone}
                            onDelete={deleteTask}
                          />
                        ))}
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          );
        }}
      </ContextTasks.Consumer>
    );
  }
}
