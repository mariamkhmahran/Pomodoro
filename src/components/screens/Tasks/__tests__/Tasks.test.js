import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import Tasks from "src/components/screens/Tasks";
import TasksContext from "src/logic/TasksContext/TasksContext.jsx";
import * as GeneralState from "src/logic/GeneralState";
import TasksData from "src/mocks/TasksData";

test("render", async () => {
  let wrapper;
  await GeneralState.set("tasks", [TasksData.data[0], TasksData.data[1]]);
  await GeneralState.set("done", [TasksData.data[4]]);
  await new Promise((resolve, reject) => {
    wrapper = renderer.create(
      <TasksContext onDidMount={inst => resolve(inst)}>
        <Tasks />
      </TasksContext>
    );
  });
  wrapper = wrapper.toJSON();
  expect(wrapper).toMatchSnapshot();
});

test("open modal", () => {
  const wrapper = mount(
    <TasksContext>
      <Tasks />
    </TasksContext>,
    {
      attachTo: document.body
    }
  );
  let isOpen = wrapper
    .find("div")
    .at(1)
    .getDOMNode()
    .getAttribute("data-is-open");
  expect(isOpen).toEqual("false");

  wrapper
    .childAt(0)
    .find("button")
    .at(1)
    .simulate("click");

  isOpen = wrapper
    .find("div")
    .at(1)
    .getDOMNode()
    .getAttribute("data-is-open");
  expect(wrapper.childAt(0).instance().state).toHaveProperty(
    "addTaskModalIsOpen",
    true
  );
  expect(isOpen).toEqual("true");
});

test("close modal", () => {
  const wrapper = mount(
    <TasksContext>
      <Tasks />
    </TasksContext>
  );
  let instance = wrapper.childAt(0).instance();
  instance.setState({ addTaskModalIsOpen: true });

  wrapper
    .find(".modal__close")
    .at(0)
    .simulate("click");
  expect(instance.state.addTaskModalIsOpen).toEqual(false);
});
