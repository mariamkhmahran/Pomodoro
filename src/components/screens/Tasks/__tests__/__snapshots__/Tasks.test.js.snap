// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<div
  className="router_output"
>
  <div
    className="modal"
    data-is-open={false}
  >
    <div
      className="modal__screen"
      onClick={[Function]}
    />
    <div
      id="snackbar"
    >
      Please fill all information..
    </div>
    <div
      className="modal__content"
    >
      <span
        className="modal__close"
        onClick={[Function]}
      >
        ×
      </span>
      <span
        className="subtitle orange"
      >
        ADD NEW TASK
      </span>
      <input
        className="inputField"
        id="name"
        placeholder="Name"
        type="text"
      />
      <input
        className="inputField"
        id="cycles"
        placeholder="Estimated total number of cycles"
        type="text"
      />
      <input
        className="inputField"
        id="deadline"
        placeholder="Deadline"
        type="date"
      />
      <div>
        <button
          className="btn addTask"
          onClick={[Function]}
        >
          Add Task
        </button>
      </div>
    </div>
  </div>
  <div
    className="subHeader"
  >
    <span
      className="subtitle"
    >
      TASKS
    </span>
    <button
      className="btn plusBtn"
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
    </button>
  </div>
  <div
    className="Main tasks"
  >
    <div>
      <div
        className="col"
      >
        <div
          className="transparent_box"
        >
          <span
            className="subtitle"
          >
            TO DO
          </span>
          <div
            className="cardList"
          >
            <div
              className="maxWidth centerAlign column"
            >
              <div
                className="task_card card"
              >
                <div
                  className="content task_content"
                >
                  <div>
                    <button
                      className="check small "
                      onClick={[Function]}
                    />
                    <div
                      className="task_data"
                    >
                      <b
                        id="name"
                      >
                        Pomodoro App
                      </b>
                      <span>
                        - Cycles done: 
                        5
                      </span>
                      <span>
                        - Estimated total number of cycles: 
                        20
                      </span>
                      <span>
                        - Deadline: 
                        1/7/2018
                      </span>
                    </div>
                    <div
                      className="actions"
                    >
                      <span
                        className="card__close"
                        onClick={[Function]}
                      >
                        ×
                      </span>
                      
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="expansionPanel maxWidth"
                id="panel"
              >
                
              </div>
            </div>
            <div
              className="maxWidth centerAlign column"
            >
              <div
                className="task_card card"
              >
                <div
                  className="content task_content"
                >
                  <div>
                    <button
                      className="check small "
                      onClick={[Function]}
                    />
                    <div
                      className="task_data"
                    >
                      <b
                        id="name"
                      >
                        Change the world
                      </b>
                      <span>
                        - Cycles done: 
                        9
                      </span>
                      <span>
                        - Estimated total number of cycles: 
                        240
                      </span>
                      <span>
                        - Deadline: 
                        1/7/2020
                      </span>
                    </div>
                    <div
                      className="actions"
                    >
                      <span
                        className="card__close"
                        onClick={[Function]}
                      >
                        ×
                      </span>
                      
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="expansionPanel maxWidth"
                id="panel"
              >
                
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="transparent_box"
        >
          <span
            className="subtitle"
          >
            DONE
          </span>
          <div
            className="cardList"
          >
            <div
              className="maxWidth centerAlign column"
            >
              <div
                className="task_card card"
              >
                <div
                  className="content task_content"
                >
                  <div>
                    <img
                      alt="checkbox"
                      className="small"
                      id="doneImg"
                      src="checked.svg"
                    />
                    <div
                      className="task_data"
                    >
                      <b
                        id="name"
                      >
                        Do Something
                      </b>
                      <span>
                        - Cycles done: 
                        12
                      </span>
                      <span>
                        - Estimated total number of cycles: 
                        10
                      </span>
                      <span>
                        - Deadline: 
                        1/1/2018
                      </span>
                    </div>
                    <div
                      className="actions"
                    >
                      <span
                        className="card__close"
                        onClick={[Function]}
                      >
                        ×
                      </span>
                      
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="expansionPanel maxWidth"
                id="panel"
              >
                
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
